hack/make/test-integration-flaky

```diff
@@ -4,12 +4,12 @@ set -e -o pipefail
 source hack/validate/.validate
 new_tests=$(
 	validate_diff --diff-filter=ACMR --unified=0 -- 'integration/*_test.go' |
-	grep -E '^(\+func )(.*)(\*testing)' || true
+		grep -E '^(\+func Test)(.*)(\*testing\.T\))' || true
 )
 
 if [ -z "$new_tests" ]; then
 	echo 'No new tests added to integration.'
-	return
+	exit 0
```

Comment:
I guess the `exit` is the problem (which exits everything ?)

Reply:
hah, I suppose so.
Let me rewrite this so it's all running in a function so we can call `return`

Reply:
Ok this should be fixed.
Now I realize how this is run... orginally thought hack/ci/janky was running it directly but no it's going through hack/make.sh first.